client/state/sites/test/reducer.js

```diff
@@ -226,6 +226,23 @@ describe( 'reducer', () => {
 			} );
 		} );
 
+		it( 'should remove an unreachable site', () => {
```

Comment:
Let's also add a test case that covers removing a site that is not present (returning original state in that case).